pgjdbc/src/test/java/org/postgresql/test/jdbc2/PGPropertyTest.java

```diff
@@ -231,6 +231,13 @@ public void testEncodedUrlValues() {
     assertEquals("password", password, PGProperty.PASSWORD.getOrDefault(parsed));
   }
 
+  @Test
+  public void testSlashPath() {
+    String databaseName = "aaa/bbb";
+    String url = "jdbc:postgresql://localhost/" + databaseName;
```

Comment:
Should the slash be URL-encoded instead?

Reply:
well that's the bigger question. 

I suspect the original reporter is using some service that automates the database name.
It appears that postgres will take the name without encoding it.

My original thought was yes it should be encoded, however the URL RFC doesn't really say anything about it. It just says that some services which are hierarchal would use the slash to delineate the hierarchy. This suggests to me that having a slash there is fine and it's up to the service how to interpret it.

Reply:
If you expect that `/` is a part of database name, then I would definitely go for encoding `/`.
I think it would make it easier to avoid ambiguities.

For instance, MySQL enforces encoding: https://dev.mysql.com/doc/connector-j/8.0/en/connector-j-reference-jdbc-url-format.html

>Any reserved characters for URLs (for example, /, :, @, (, ), [, ], &, #, =, ?, and space) that appear in any part of the connection URL must be percent encoded.

Reply:
works for me